src/screens/ProposalDetail.js

```diff
@@ -71,6 +60,18 @@ function ProposalDetail({
     executed,
   } = proposal
 
+  const hasCancelRole = useMemo(() => {
+    if (!connectedAccount) {
+      return false
+    }
+    return permissions.find(
+      ({ roleHash, granteeAddress }) =>
+        (roleHash === CANCEL_ROLE_HASH &&
+          addressesEqual(granteeAddress, connectedAccount)) ||
+        addressesEqual(creator, connectedAccount)
+    )
+  }, [connectedAccount, creator, permissions])
```

Comment:
Would  refactor to something like:

```suggestion
  const canCancelProposal = useMemo(() => {
    if (!connectedAccount) {
      return false
    }
    
    if (addressesEqual(creator, connectedAccount)) {
      return true
    }
    
    return permissions.find(
      ({ roleHash, granteeAddress }) =>
        (roleHash === CANCEL_ROLE_HASH &&
          addressesEqual(granteeAddress, connectedAccount))
    )
  }, [connectedAccount, creator, permissions])
```